### PR TITLE
Add infinite scrolling behavior

### PR DESCRIPTION
**Pull request by Pikaju**
The canvas can be scrolled infinitely in any direction by dragging the background with the left mouse button. No browser scrollbars are required anymore.

### CODE REVIEW

File: src/components/Canvas.vue [hunk 1]
```diff
@@ -1,40 +1,55 @@
...
-        ref="nodes"
-        @connect="onConnect"
-        @move="onNodeMove"
-      >
-        <component :is="node.component" />
-      </p-node>
+  <div id="canvas">
+    <p-toolbox id="toolbox" :types="nodeTypes" @add="addNode" />
+    <p-background v-touch-pan.mouse.prevent="handlePan" :scroll="scroll">
+      <div :style="{position: 'absolute', top: -scroll.y + 'px', left: -scroll.x + 'px'}">
```
Comment (AKoll) [inline src/components/Canvas.vue]: Inserting spaces to seperate the object curly braces might be more consistent in terms of codestyle. Have you done it this way to be more compact?

File: src/components/Background.vue [hunk 2]
```diff
@@ -1,17 +1,29 @@
 <template>
-  <div id="background" class="grid-background">
-    <slot/>
+  <div
+    id="background"
+    class="grid-background"
+    :style="{backgroundPosition: `${-scroll.x}px ${-scroll.y}px`}"
```
Comment (AKoll) [inline src/components/Background.vue]: Inserting spaces to seperate the object curly braces might be more consistent in terms of codestyle. Have you done it this way to be more compact?

File: src/components/Canvas.vue (hunk 1 above)
Comment (Pikaju) [reply src/components/Canvas.vue]: My bad. I thought our linter settings disallowed spaces around object parentheses.

**Comment by LukaZdr** [comment]
also erstmal ist die idee kompletter mist
https://de.wikipedia.org/wiki/Mist
hier ein beispiel

**Comment by Pikaju** [comment]
> This works nicely, however there needs to be some indication as to where all the nodes are located.
> Otherwise, nodes will get lost on the canvas and potentially lead to performance loss.

Do you have a proposal as to how to design such an indicator? As always, I would like to point out that this may not necessarily be the purpose of this particular branch.

**Comment by ProtoNXT** [comment]
As a first idea I thought of a minimap, that allows to view a large section of the canvas using only little screen space and also acts as a navigation aid. Another option would be to make the canvas zoomable, so that the graphs structure can be more easily grasped.
Those approaches will certainly not solve all problems, but should speed up the workflow for larger projects.

**Pull request merged**